resources/views/markdownField.blade.php

```diff
@@ -20,124 +20,77 @@ function debounce(func, timeout = 300) {
+                    }
+
+                    $wire.upload(`componentFileAttachments.{{ $getStatePath() }}`, file, () => {
+                        $wire.getComponentFileAttachmentUrl('{{ $getStatePath() }}').then((url) => {
                             if (! url) {
-                                return
+                                return onError('File could not be uploaded');
                             }
 
-                            alert('file uploaded to ' + url);
-
-                            {{--
-                            $refs.imageTrigger.click()
-
-                            const urlStart = $refs.textarea.selectionStart + 2
-                            const urlEnd = urlStart + 3
-
-                            state = [
-                                $refs.textarea.value.substring(0, urlStart),
-                                url,
-                                $refs.textarea.value.substring(urlEnd)
-                            ].join('')
-
-                            $refs.textarea.selectionStart = urlStart - 2
-                            $refs.textarea.selectionEnd = urlStart - 2
-
-                            render()
-
-                            --}}
+                            onSuccess(url);
                         })
-
-                        {{--
-                        const data = new FormData();
-                        data.append('file', file);
-
-                        fetch('{{ action(\Spatie\Mailcoach\Http\Api\Controllers\UploadsController::class) }}', {
-                            method: 'POST',
-                            body: data,
-                            credentials: 'same-origin',
-                            headers: {
-                                'X-CSRF-Token': '{{ csrf_token() }}',
-                            },
-                        })
-                            .then(response => response.json())
-                            .then(({ success, file }) => {
-                                if (! success) {
-                                    return onError();
-                                }
-
-                                onSuccess(file.url);
-                            });
-                    --}}
-                    },
+                    });
+                },
             });
 
-            editor.codemirror.on("change", debounce(() => {
-                this.markdown = editor.value();
-                this.$refs.editor.dirty = true;
+            editor.codemirror.on('change', debounce(() => {
+                state = editor.value();
             }));
-        }
-    </script>
 
-    <div
-        x-data="{ state: $wire.entangle('{{ $getStatePath() }}').defer }"
-        x-init="
-            editor = new EasyMDE({ element: $refs.textarea })
-            editor.value(state ?? '')
-            editor.codemirror.on('change', () => state = editor.value())
             $watch('state', () => {
```

Comment:
This watcher will be triggered when the state changes from the backend, for example when `$set()` is called